Lib/sqlite3/test/dbapi.py

```diff
@@ -170,26 +170,35 @@ def test_in_transaction_ro(self):
         with self.assertRaises(AttributeError):
             self.cx.in_transaction = True
 
+
+class OpenTests(unittest.TestCase):
+    def tearDown(self):
+        unlink(TESTFN)
+
     def test_open_with_path_like_object(self):
         """ Checks that we can successfully connect to a database using an object that
             is PathLike, i.e. has __fspath__(). """
-        self.addCleanup(unlink, TESTFN)
         class Path:
             def __fspath__(self):
                 return TESTFN
         path = Path()
         with sqlite.connect(path) as cx:
             cx.execute('create table test(id integer)')
+        cx.close()
```

Comment:
Maybe a helper context manager would be helpful? It would close the DB and then remove the file. Something like:

```
@contextlib.contextmanager
def connect(path):
  conn = sqlite.connect(path)
  try:
    yield conn
  finally:
    conn.close()
    unlink(path)
```

I'm not sure if it's worth it if there are just a few users of such helper function. Your PR is also good like that.

Reply:
I'm not sure it's worth it, but I'll keep it in mind if we add more such users.

Reply:
I changed my mind; the helper is nice. I'll add it :)

Reply:
See aba8dd724618df1a3ab2cad90b03436cb9bc9fd8